docs/service-clients.md

```diff
@@ -0,0 +1,92 @@
+# Service Clients
+
+Do you wish to allow users to authenticate to your app, as well as fetch
+some information about them? Then you don't want to be a Service Client.
+Be a regular client.
+
+Service Clients exist as privileged apps that need to be able to request
+information about a user, without ever having received permission to do
+so. Sounds nefarious. However, as said, these are **privileged** apps,
+meaning they are also run by **us** (Mozilla). So these apps are not
+gaining access to something that we don't already know. Plus, before
+ever being promoted to a Service Client, real humans are involved to
+make sure it's the correct use case.
+
+## All Powerful
+
+A Service Client exists by being in the config array `serviceClients`.
+Each entry (if any) is an object, with the following values:
+
+- `id` - `String`: a unique hex value in identical format to regular
+  client ids.
+- `name` - `String`: a plaintext name for the client that's friendly to
+  human readers.
+- `scope` - `String`: space-separated scopes that this client will be
+  using when requesting access tokens.
+- `jku` - `String`: a unique URL that will host a JWK Set. Unique as in
+  unique in the current list of Service Clients.
+
+## JKUs and JWK Sets
+
+Imagine a service client with the `jku` of `https://example.dom.ain/keys`.
+The document hosted at that URL should contain something like:
+
+```json
+{
+  "keys":[{
+    "kid":"key-id-can-whatever-1",
+    "use": "sig",
+    "kty":"RSA",
+    "n":"W_lCUvksZMVxW2JLNtoyPPshvSHng28H5FggSBGBjmzv3eHkMgRdc8hpOkgcPwXYxHdVM6udtVdXZtbGN8nUyQX8gxD3AJg-GSrH3UOsoArPLCmcxwIEpk4B0wqwP68oK8dQHt0iK3N-XeCnMpv75ULlVn3LEOZT8CsuNraVOthYeClUb8r1PjRwqRB06QGNqnnhcPMmh-6cRzQ9HmTMz6CDcugiH5n2sjrvpeBugEsnXt3KpzVdSc4usXrIEmLRuFjwFbkzoo7FiAtSoXxBqc074qz8ejm-V0-2Wv3p6ePeLODeYkPQho4Lb1TBdoidr9RHY29Out4mhzb4nUrHHQ",
+    "e":"AQAB"
+  }]
+}
+```
+
+This JKU is important when making requests for access tokens.
+
+## JWTs and Authorization
+
+To request an access token, a service client must generate a signed [JWT][],
+and then send it to our [/v1/token][] endpoint.
+
+Here's an example of creating a JWT in JavaScript:
+
+```js
+var now = Math.floor(Date.now() / 1000); // in seconds
+var header = {
+  alg: 'RS256',
+  typ: 'JWT',
+  jku: 'https://basket.accounts.firefox.com/.well-known/jku',
+  kid: 'k1'
+};
+var claims = {
+  scope: 'profile:email',
+  aud: 'https://oauth.accounts.firefox.com/v1/token',
+  iat: now,
+  exp: now + (60 * 5),
+  sub: userId + '@accounts.firefox.com'
```

Comment:
nit: I think this will have to be "@api.accounts.firefox.com" in production, which is pretty ugly, but it's what we have...